tools/ci-cdk/canary-runner/src/run.rs

```diff
@@ -10,6 +10,9 @@
 //
 // Also consider using the `AWS_PROFILE` and `AWS_REGION` environment variables
 // when running this locally.
+//
+// CAUTION: This subcommand will `git reset --hard` in some cases. Don't ever run
+// it against a smithy-rs repo that you're actively working in.
```

Comment:
I think it's low risk since testing this tool requires checking out smithy-rs inside of aws-sdk-rust, and that's the smithy-rs it would be running this on. I just wanted to call it out as something destructive that would happen.